src/openrct2/network/Network.cpp

```diff
@@ -665,6 +666,19 @@ uint8_t Network::GetPlayerID()
     return player_id;
 }
 
+NetworkConnection* Network::GetPlayerConnection(uint8_t id)
+{
+    auto player = GetPlayerByID(id);
+    if (player != nullptr)
+    {
+        return std::find_if(
+                   client_connection_list.begin(), client_connection_list.end(),
+                   [player](const auto& conn) -> bool { return conn->Player == player; })
+            ->get();
```

Comment:
```suggestion
        auto clientIt = std::find_if(
                   client_connection_list.begin(), client_connection_list.end(),
                   [player](const auto& conn) -> bool { return conn->Player == player; });
        return clientIt != client_connection_list.end() ? clientIt->get() : nullptr;
```
If it doesn't find the player, it won't try to dereference `std::end`